allensdk/brain_observatory/behavior/data_objects/running_speed/running_acquisition.py

```diff
@@ -83,79 +65,52 @@ def __init__(
         self._stimulus_timestamps = stimulus_timestamps
 
     @classmethod
-    @cached(cache=LRUCache(maxsize=10), key=from_json_cache_key)
-    def from_json(
-        cls,
-        dict_repr: dict,
-    ) -> "RunningAcquisition":
-        stimulus_file = BehaviorStimulusFile.from_json(dict_repr)
-        stimulus_timestamps = StimulusTimestamps.from_json(dict_repr=dict_repr,
-                                                           monitor_delay=0.0)
+    def from_stimulus_file(
+            cls,
+            behavior_stimulus_file: BehaviorStimulusFile,
+            sync_file: Optional[SyncFile] = None) -> "RunningAcquisition":
+
+        if sync_file is not None:
+            stimulus_timestamps = StimulusTimestamps.from_sync_file(
```

Comment:
Why don't we use the `StimulusTimestamps.from_multiple_stimulus_blocks` for running speed?

Reply:
When we're generating the filtered RunningSpeed, we need to filter each stimulus block separately, generating a unique data frame for the behavior, mapping, and replay blocks before concatenating the data frame. I didn't want to add the complication of creating a monolith of timestamps and then breaking it up into distinct chunks for behavior, mapping, and replay.

`StimulusTimestamps.from_multiple_stimulus_blocks` is intended for something like licks, where we really are only generating a dataframe from one block (the behavior stimulus block), in which case you would run it with a non-`None` `stims_of_interest`.

This is not the most well-motivated reasoning I've ever engaged in. Really, this implementation is the most direct path from what we were given. We can revisit this decision when things calm down. Though, given all of the complications surrounding what data includes a monitor delay and what data does not, I am becoming less convinced that we should be passing around `StimulusTimestamps`. I think I'm in favor of passing around stimulus and sync files and letting each data object call the appropriate combination of functions to assemble their own timestamps. That is only an opinion, though.